src/patch/apply.js

```diff
@@ -89,7 +90,7 @@ export function applyPatch(source, uniDiff, options = {}) {
       let line = hunk.lines[j],
           operation = (line.length > 0 ? line[0] : ' '),
           content = (line.length > 0 ? line.substr(1) : line),
-          delimiter = hunk.linedelimiters[j];
+          delimiter = assumeLineDelimiter || hunk.linedelimiters[j];
```

Comment:
Hmm... this should be the other way round - i.e. `hunk.linedelimiters[j] || assumeLineDelimiter` - shouldn't it? As currently written, the line delimiter from `options` will take precedence over one actually provided in the hunk.